File: mine/mineea/src/main/cpp/ta/trader/td0001/TaMa2OrderUpdator.cpp
#include "TaMa2OrderUpdator.h"

TaMa2OrderUpdator::TaMa2OrderUpdator(STRING pName, TaOrderExplainer* pExp) :
		TaStage2OrderUpdator(pName, pExp) {
	this->twoMa = new TaTwoMovingAverageTimeBar(20, 100);
}
TaMa2OrderUpdator::~TaMa2OrderUpdator(){
	delete this->twoMa;
}
double TaMa2OrderUpdator::calculateFirstProfitForStage2(double stoploss2){

	double rt = TaLang::__iATR(NULL,TaLang::PERIOD_D1_,10,0);
	rt = rt * 5.0;
	return rt;
}
double TaMa2OrderUpdator::retriveOriginalTakeProfitRate(){
	return TaConstants::OST_MA2_TAKE_PROFIT_RATE;
}

double TaMa2OrderUpdator::retriveOriginalLots(){
	return TaConstants::OST_MA2_ORIGINAL_LOTS;
}

double TaMa2OrderUpdator::calculateStopLossForStage2(){

	return this->twoMa->getMidValue(0);

}

File: mine/tas/src/main/java/org/ta/trader/speculate/ma2/TaMa2OrderUpdator.java
package org.ta.trader.speculate.ma2;

import org.ta.trader.hista.syriod.TaSyriodWindow;
import org.ta.trader.hista.tohlcv.TaTwoMovingAverageTimeBar;
import org.ta.trader.speculate.TaStage2OrderUpdator;

public class TaMa2OrderUpdator extends TaStage2OrderUpdator {
	private TaTwoMovingAverageTimeBar twoMa;

	public TaMa2OrderUpdator(TaSyriodWindow window) {
		super(window);
		this.twoMa = new TaTwoMovingAverageTimeBar(20, 100, window);
	}

	@Override
	protected double calculateFirstProfitForStage2(double stoploss2) {

		double rt = 0.00100;// TODO TaLang::__iATR(NULL,TaLang::PERIOD_D1_,10,0);
		rt = rt * 5.0;
		return rt;
	}

	@Override
	protected double calculateStopLossForStage2() {

		return this.twoMa.getMidValue(0);

	}
}
